src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -364,11 +365,11 @@ export class ConnectionDialogService implements IConnectionDialogService {
 	}
 
 	private handleFillInConnectionInputs(connectionInfo: IConnectionProfile): void {
-		this._connectionManagementService.addSavedPassword(connectionInfo).then(connectionWithPassword => {
+		this._connectionManagementService.addSavedPassword(connectionInfo).then(async connectionWithPassword => {
```

Comment:
Nit: This call to addSavedPassword can be avoided since createModel will take care of it. But even if this exists, it should be skipped second time as password will be populated.. so functionally we're fine. 

For readability, I'd suggest cleaning it up to handle password addition only in 1 place.